ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/reloperators/HiveTableScan.java

```diff
@@ -202,6 +203,12 @@ public HiveTableScan copyIncludingTable(RelDataType newRowtype) {
   // Also include partition list key to trigger cost evaluation even if an
   // expression was already generated.
   @Override public RelWriter explainTerms(RelWriter pw) {
+    if (pw instanceof RelTreeSignatureWriter) {
+      return super.explainTerms(pw)
+          .item("tableScanTrait", this.tableScanTrait)
+          .itemIf("fromVersion", ((RelOptHiveTable) table).getHiveTableMD().getVersionIntervalFrom(),
+              isNotBlank(((RelOptHiveTable) table).getHiveTableMD().getVersionIntervalFrom()));
```

Comment:
HiveTableScan doesn't retain the equivalents of `TableScanDesc#getPredicateString` or `TableScanDesc#getRowLimit`.
So, we can't unify ASTNodes or RelNodes based on `RelTreeSignature#equals`. Otherwise, Operators which should not belong to the same group will be unified later.